vpr/src/route/rr_graph_fpga_interchange.cpp

```diff
@@ -0,0 +1,1877 @@
+#include <kj/std/iostream.h>
+#include <zlib.h>
+#include <algorithm>
+#include <map>
+#include <queue>
+
+#include "rr_graph_fpga_interchange.h"
+
+#include "arch_util.h"
+#include "vtr_time.h"
+#include "vtr_error.h"
+#include "globals.h"
+#include "check_rr_graph.h"
+#include "rr_metadata.h"
+#include "rr_graph_indexed_data.h"
+#include "fpga_interchange_arch_utils.h"
+
+//#define DEBUG
+
+using namespace DeviceResources;
+using namespace LogicalNetlist;
+using namespace capnp;
+
+// Helper types definitions
+
+// t_location in the grid: <x, y>
+typedef std::pair<int, int> t_location;
+
+// edge between two wires: <switch_id, tile_id, <name, wire0, wire1, forward>>
+typedef std::tuple<int, int, std::tuple<int, int, int, bool>> t_pip;
+
+// Helper enumerations
+
+enum constant {
+    GND = 0,
+    VCC,
+};
+
+enum node_sides {
```

Comment:
Comment what these are used for.  Side of a block that an rr_node is on?  Or something else?